provider/resources.go

```diff
@@ -2192,6 +2192,16 @@ compatibility shim in favor of the new "name" field.`)
 					"node_group_name": tfbridge.AutoName("nodeGroupName", 255, "-"),
 				},
 			},
+			"aws_eks_cluster": {
+				TransformFromState: func(_ context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
```

Comment:
I might be missing something, but why is the State Upgrader not working here? Upstream configures it here: 
- https://github.com/hashicorp/terraform-provider-aws/blob/4b740e8f8919ff1b4571e15b286fc574456bd56d/internal/service/eks/cluster_migrate.go#L260
- https://github.com/hashicorp/terraform-provider-aws/blob/4b740e8f8919ff1b4571e15b286fc574456bd56d/internal/service/eks/cluster.go#L50

Reply:
Ah, great find! This is likely https://github.com/pulumi/pulumi-terraform-bridge/pull/2081

Reply:
That is still not rolled out to AWS. We could roll it out together with PRC.

Reply:
Can we opt in specific resources only to https://github.com/pulumi/pulumi-terraform-bridge/pull/2081?

I think both are great enhancements, but rolling both out at the same time makes it harder to reason about potential issues that come up

Reply:
No we can't filter which resources to enable https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 for.

I have confirmed that indeed that fixes the EKS replacement.

Reply:
I'd vote for rolling them out separately. https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 Seems relatively straight forward to roll out (famous last words), so wdyt about doing that right away?

Reply:
I am unsure of the effects of https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 without PRC - we have not tested that anywhere, so I'd be hesitant to do that.

Reply:
There were concerns that 2081 might uncover latent panics which were hidden. The non-PRC bridge is known to transform some properties incorrectly, so running upgrades which were not run before could conceivably uncover some issues.

Reply:
If rolling the two together is a big concern, we could fix the EKS cluster as in this PR.

That might cause similar issues in other resources though.